clippy_dev/src/fmt.rs

```diff
@@ -312,27 +298,18 @@ fn run_rustfmt(clippy: &ClippyInfo, update_mode: UpdateMode) {
             cmd.stdout(Stdio::null())
                 .stdin(Stdio::null())
                 .stderr(Stdio::piped())
-                .args(["--config", "show_parse_errors=false"]);
+                .args(["--unstable-features", "--skip-children"]);
             cmd
         },
-        |cmd| match cmd.spawn() {
-            Ok(x) => children.push(("rustfmt", x)),
-            Err(ref e) => panic_action(&e, ErrAction::Run, "rustfmt".as_ref()),
-        },
-        WalkDir::new("tests")
-            .into_iter()
-            .filter_entry(|p| p.path().file_name().is_none_or(|x| x != "skip_rustfmt"))
-            .filter_map(|e| {
-                let e = e.expect("error reading `tests`");
-                e.path()
-                    .as_os_str()
-                    .as_encoded_bytes()
-                    .ends_with(b".rs")
-                    .then(|| e.into_path().into_os_string())
-            }),
-    );
+        args.iter(),
+    )
+    .map(|mut cmd| match cmd.spawn() {
+        Ok(x) => x,
+        Err(ref e) => panic_action(&e, ErrAction::Run, "rustfmt".as_ref()),
```

Comment:
It would be more ergonomic to have `panic_action()` take a `impl AsRef<Path>` (and call `.as_ref()` in it) and to use `&str` directly:

```suggestion
        Err(e) => panic_action(&e, ErrAction::Run, "rustfmt"),
```